src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -0,0 +1,97 @@
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import net.sf.saxon.type.Type;
+
+/**
+ * Represents element node of Xpath-tree.
+ *
+ * @author Timur Tibeyev
+ */
+public class ElementNode extends AbstractNode {
+    /** Represents value of TokenTypes#IDENT. */
+    private String image;
+
+    /**
+     * Creates a new {@code ElementNode} instance.
+     *
+     * @param tokenType the number representing constant from {@code TokenTypes}
+     * @param parent {@code Node} parent of the current node
+     * @param document {@code Node} root of the tree
+     * @param detailNode reference to {@code DetailNode}
+     * @param detailAST reference to {@code DetailAST}
+     */
+    public ElementNode(int tokenType, Node parent, Node document, DetailNode detailNode, DetailAST detailAST) {
+        super(tokenType);
+        this.parent = parent;
+        this.document = document;
+        this.detailNode = detailNode;
+        this.detailAST = detailAST;
+
+        if (detailAST.branchContains(TokenTypes.IDENT)) {
+            this.image = detailAST.findFirstToken(TokenTypes.IDENT).getText();
+        }
+
+        if (parent != null) {
+            ((AbstractNode) parent).addChild(this);
+        }
+    }
+
+    /**
+     * Returns type of the node.
+     * @return node kind
+     */
+    public int getNodeKind() {
+        return Type.ELEMENT;
+    }
+
+    /**
+     * Returns line number.
+     * @return line number
+     */
+    public int getLineNumber() {
+        return detailAST.getLineNo();
+    }
+
+    /**
+     * Returns column number.
+     * @return column number
+     */
+    public int getColumnNumber() {
+        return detailAST.getColumnNo();
+    }
+
+    /**
+     * Returns local part.
+     * @return local part
+     */
+    public String getLocalPart() {
+        return detailAST.getText();
+    }
+
+    /**
+     * Returns string value.
+     * @return string value
+     */
+    public String getStringValue() {
+        return detailAST.getText();
+    }
+
+    /**
+     * Returns display name.
+     * @return display name
+     */
+    public String getDisplayName() {
+        return detailAST.getText();
+    }
+
+    public String getAttributeValue(String s, String s1) {
```

Comment:
what does `s1` mean? Why it is so special? What is the difference between `s` and `s1`? 
The names of the parameters should be changed to explain their purposes.